### ISSUE TITLE
Compass problem in V3.0.1R2

### ISSUE THREAD

**Issue by Aussi1994** [MegaPirateNG/ardupilot-mpng#2]
Greetings,
I cannot seem to get the compass to work properly in version 3.0.1R2. The yaw angle does not reflect the correct heading, it just slowly drifts. Everything was working fine with version 2.8R3 loaded. As a test, I reloaded version 2.8R3 and everything worked. The hardware is the RCTIMER CRIUS AIOP V2 board.

Here is a list of some of the compass parameters and values:
COMPASS_LEARN = 0
COMPASS_USE = 1
COMPASS_ORIENT = 0
COMPASS_AUTODEC = 1
COMPASS_OFS_X = 275
COMPASS_OFS_Y = 100
COMPASS_OFS_Z = 340

Any help would be appreciated,

Bob

**Comment by SirAlex** [comment]
I'm suggest you to disable AutoDeclination feature and manually enter declination. Then you must make compass calibration. If nothing help, try to enter test -> compass command in terminal, check the compass works in it.
Also do not use parameters from 2.8 or 2.9. After flashing 3.0.1, goto terminal and do setup -> erase, and after reboot, make all calibrations again.

**Comment by c-bob** [comment]
Same here!
Compass is not doing well!
2.9x -> no problems, works like a charm. Loiter was nearly perfect! RTL the same.
Now:
- loiter: toilet-bowl (cut it at a radius of more then around 20m) 
- RTL not working!
  Will try it with manual declination - but now rain starts :/

**Comment by SirAlex** [comment]
@c-bob , It's not problem in MPNG, Arducopter 3.0.1 uses different algorithms than in previous versions. Please, read notes for ArduCopter 3.0.1, especially warning N1:
http://diydrones.ning.com/forum/topics/arducopter-3-0-1-released

**Comment by c-bob** [comment]
@SirAlex , well I know that it isnt a problem of mpng. I just think it's a pity that it looks now as if it would be a backward step.
Even with a small impact of current to the compass it was working very pretty in the last versions.

**Comment by SirAlex** [comment]
@c-bob , Just make compassmot calibration. It will reduce compass interference and will remove toilet-bowling.
Please, read instructions here: http://copter.ardupilot.com/wiki/ac_compasssetupupadvanced/#Compassmot_8211_compensation_for_interference_from_the_PDB_ESC_motors

**Comment by c-bob** [comment]
@SirAlex , Well ,thanks for the link. But I did it before I noticed the auto-declination issue...
So I'll try it again and let you now, when my weather conditions are better - maybe later the day ;)

**Comment by Aussi1994** [comment]
Hi,
I disabled AutoDeclination and performed numerous compass live calibrations. I was not successful in get all three parameters within the normal limits, only compass_ofs_y was valid. This was also true when I was using version 2.8R3, I had to manually figure out the offsets to make things work. 

After failing that, I tried a compass test from the CLI with the following results shown:
-1
0
Compass initialization failed

Does the compass test even work? I read it had some problems in the past..

http://diydrones.com/forum/topics/mission-planner-terminal-test-compass?commentId=705844%3AComment%3A1140665&xg_source=activity

Best regards,

Bob
